packages/framework/src/Component/Router/DomainRouterFactory.php

```diff
@@ -49,28 +52,80 @@ class DomainRouterFactory
      */
     protected $requestStack;
 
+    /**
+     * @var \Symfony\Component\DependencyInjection\ContainerInterface|null
+     */
+    protected ?ContainerInterface $container;
+
+    /**
+     * @var string|null
+     */
+    protected ?string $cacheDir;
+
     /**
      * @param mixed $routerConfiguration
-     * @param \Symfony\Component\Config\Loader\LoaderInterface $configLoader
+     * @param \Symfony\Component\Config\Loader\LoaderInterface|null $configLoader
      * @param \Shopsys\FrameworkBundle\Component\Router\LocalizedRouterFactory $localizedRouterFactory
      * @param \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlRouterFactory $friendlyUrlRouterFactory
      * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
      * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+     * @param \Symfony\Component\DependencyInjection\ContainerInterface|null $container
+     * @param string|null $cacheDir
      */
     public function __construct(
         $routerConfiguration,
-        LoaderInterface $configLoader,
+        ?LoaderInterface $configLoader,
         LocalizedRouterFactory $localizedRouterFactory,
         FriendlyUrlRouterFactory $friendlyUrlRouterFactory,
         Domain $domain,
-        RequestStack $requestStack
+        RequestStack $requestStack,
+        ?ContainerInterface $container = null,
+        ?string $cacheDir = null
```

Comment:
Why is it passed there? It can be taken from the container directly.

Reply:
It can be taaken from container, as is in fallback in costructor. I just do not like the way I am just picking some configuration argument directly from container an want to have it strictly passed via constructor.

Reply:
Almost seems like ParameterBag here could solve it nicely.

Reply:
You are right. To me it is like an antipattern which I would love to avoid when possible.

Reply:
Okay, I see. Just a final thought about this – does it make sense to swap the order of the arguments? 
First $cacheDir and container as the last. Because in the next major we probably remove the dependency on the container as it's not necessary when $cacheDir is provided and it seems to me it will be a little more clear in the future upgrade notes

Reply:
Not really. The container is required and also will be in the future as a dependency for `Symfony\Bundle\FrameworkBundle\Routing\Router`.

Reply:
My bad, thanks for explanation